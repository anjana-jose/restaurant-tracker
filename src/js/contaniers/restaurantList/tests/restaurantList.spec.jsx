/* eslint-disable react/prop-types */
/* eslint-disable react/no-multi-comp */

import React from 'react';
import renderer from 'react-test-renderer';

const { RestaurantList } = require('../restaurantList');

describe('containers/restaurantList', () => {
  it('without data and cities', () => {
    const data = {};
    const cities = [];

    const component = renderer.create(
      <RestaurantList data={data} cities={cities} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('With out url and user id', () => {
    const data = { id: 'id1', name: 'restaurant1' };
    const cities = ['toronto', 'Mississauga'];

    const component = renderer.create(
      <RestaurantList data={data} cities={cities} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
