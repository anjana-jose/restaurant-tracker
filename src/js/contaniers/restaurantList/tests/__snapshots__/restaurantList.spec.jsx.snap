// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/restaurantList With out url and user id 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "height": "calc(100% - 60px)",
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "height": "100%",
        "marginLeft": "10px",
        "width": "15%",
      }
    }
  >
    <h4
      style={
        Object {
          "color": "#5b616b",
          "marginBottom": "5px",
        }
      }
    >
      <img
        alt=""
        src="https://careers.just-eat.com/assets/images/icons/search.png"
        style={
          Object {
            "height": "14px",
            "marginRight": "6px",
            "marginTop": "2px",
          }
        }
      />
      <span
        style={
          Object {
            "color": "#5b616b",
          }
        }
      >
        Refine
      </span>
      <label
        htmlFor="querySearch"
      >
        <input
          id="querySearch"
          onChange={[Function]}
          style={
            Object {
              "width": "95%",
            }
          }
          type="text"
        />
      </label>
    </h4>
    <h4
      style={
        Object {
          "color": "#5b616b",
          "marginBottom": "5px",
        }
      }
    >
      City Selected:
      <span
        style={
          Object {
            "color": "#5b616b",
            "marginLeft": "5px",
            "textTransform": "capitalize",
          }
        }
      >
        toronto
      </span>
      <label
        htmlFor="search-city"
      >
        <input
          id="search-city"
          onChange={[Function]}
          style={
            Object {
              "width": "95%",
            }
          }
          type="text"
        />
      </label>
    </h4>
    <div
      style={
        Object {
          "height": "calc(100% - 130px",
          "overflow": "auto",
        }
      }
    >
      <div>
        <input
          id="radio-button"
          name="city"
          onChange={[Function]}
          type="radio"
          value="toronto"
        />
        <span
          style={
            Object {
              "color": "#5b616b",
              "fontSize": "16px",
              "fontWeight": "normal",
            }
          }
        >
          toronto
        </span>
      </div>
      <div>
        <input
          id="radio-button"
          name="city"
          onChange={[Function]}
          type="radio"
          value="Mississauga"
        />
        <span
          style={
            Object {
              "color": "#5b616b",
              "fontSize": "16px",
              "fontWeight": "normal",
            }
          }
        >
          Mississauga
        </span>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "height": "100%",
        "overflow": "auto",
        "width": "85%",
      }
    }
  >
    <table
      style={
        Object {
          "border": "1px solid",
          "width": "100%",
        }
      }
    >
      <tr
        style={
          Object {
            "backgroundColor": "#112e51",
            "border": "1px solid",
            "color": "#ffffff",
            "height": 50,
          }
        }
      >
        <th>
          Name
        </th>
        <th>
          Area
        </th>
        <th>
          Address
        </th>
        <th>
          Phone
        </th>
        <th>
          Photo
        </th>
      </tr>
      <div>
        Loading data...
      </div>
    </table>
  </div>
</div>
`;

exports[`containers/restaurantList without data and cities 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "height": "calc(100% - 60px)",
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "height": "100%",
        "marginLeft": "10px",
        "width": "15%",
      }
    }
  >
    <h4
      style={
        Object {
          "color": "#5b616b",
          "marginBottom": "5px",
        }
      }
    >
      <img
        alt=""
        src="https://careers.just-eat.com/assets/images/icons/search.png"
        style={
          Object {
            "height": "14px",
            "marginRight": "6px",
            "marginTop": "2px",
          }
        }
      />
      <span
        style={
          Object {
            "color": "#5b616b",
          }
        }
      >
        Refine
      </span>
      <label
        htmlFor="querySearch"
      >
        <input
          id="querySearch"
          onChange={[Function]}
          style={
            Object {
              "width": "95%",
            }
          }
          type="text"
        />
      </label>
    </h4>
    <h4
      style={
        Object {
          "color": "#5b616b",
          "marginBottom": "5px",
        }
      }
    >
      City Selected:
      <span
        style={
          Object {
            "color": "#5b616b",
            "marginLeft": "5px",
            "textTransform": "capitalize",
          }
        }
      >
        toronto
      </span>
      <label
        htmlFor="search-city"
      >
        <input
          id="search-city"
          onChange={[Function]}
          style={
            Object {
              "width": "95%",
            }
          }
          type="text"
        />
      </label>
    </h4>
    <div
      style={
        Object {
          "height": "calc(100% - 130px",
          "overflow": "auto",
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "height": "100%",
        "overflow": "auto",
        "width": "85%",
      }
    }
  >
    <table
      style={
        Object {
          "border": "1px solid",
          "width": "100%",
        }
      }
    >
      <tr
        style={
          Object {
            "backgroundColor": "#112e51",
            "border": "1px solid",
            "color": "#ffffff",
            "height": 50,
          }
        }
      >
        <th>
          Name
        </th>
        <th>
          Area
        </th>
        <th>
          Address
        </th>
        <th>
          Phone
        </th>
        <th>
          Photo
        </th>
      </tr>
      <div>
        Loading data...
      </div>
    </table>
  </div>
</div>
`;
